lib/logstash/inputs/imap.rb

```diff
@@ -41,6 +58,28 @@ class LogStash::Inputs::IMAP < LogStash::Inputs::Base
   # Path to file with last run time metadata
   config :sincedb_path, :validate => :string, :required => false
 
+  def initialize(*params)
+    super
+
+    if original_params.include?('headers_target')
+      @headers_target = normalize_field_ref(@headers_target)
+    else
+      @headers_target = '[@metadata][input][imap][headers]' if ecs_compatibility != :disabled
+    end
+
+    if original_params.include?('attachments_target')
+      @attachments_target = normalize_field_ref(@attachments_target)
+    else
+      @attachments_target = ecs_compatibility != :disabled ? '[@metadata][input][imap][attachments]' : '[attachments]'
+    end
+  end
+
+  def normalize_field_ref(target)
+    # so we can later event.set("#{target}[#{name}]", ...)
```

Comment:
nit: not strictly necessary, but perhaps more efficient at runtime to have had pre-normalized.